crates/solidity/outputs/cargo/tests/src/parser/mod.rs

```diff
@@ -0,0 +1,29 @@
+use semver::Version;
+use slang_solidity::{kinds::RuleKind, language::Language};
+
+#[test]
+fn test_precedence_expression() {
```

Comment:
I suggest moving these tests to the existing snapshot testing, as it covers all of these checks, but they don't need to be manually updated.

Reply:
This tests the `parse(RuleKind::SomePrecedenceExpression)` public API, which is not executed when parsing in general, as it's usually constructed/reduced by the main precedence parser.

With this, we ensure that the verification logic works correctly for the precedence expression; otherwise we would not have coverage for it.

Reply:
Yes, but with your PR, we can finally test it, right?
This enables adding snapshot tests under `cst_output/SomePrecedenceExpression/test_name`, where we previously couldn't.

Reply:
I somehow missed that the CST test harness automatically calls `.parse(RuleKind::from_str(...))` directly, sorry about that. In that case it makes sense, sure! I'll update that in a sec.

Reply:
Done in https://github.com/NomicFoundation/slang/pull/699/commits/fe12dd3bd763ec4cdb90b7945e09dc268d116156; also moved some tests like the exponentiation expression or the conditional expression into their dedicated parser category since we were getting repetitive with the prefixes a bit and these make sense to have their own category rather than piggy-backing on being an expression and parsing it as such.